lib/IntermediateRepresentation/afforestation.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FORTRAN_INTERMEDIATEREPRESENTATION_AFFORESTATION_H_
+#define FORTRAN_INTERMEDIATEREPRESENTATION_AFFORESTATION_H_
+
+#include <string>
+
+namespace Fortran::parser {
+class Program;
+}
+
+namespace Fortran::semantics {
+class SemanticsContext;
+}
+
+namespace Fortran::IntermediateRepresentation {
+class Program;
+
+void SetDebugChannel(const std::string &filename);
```

Comment:
Maybe SemanticsContext should contain a `std::ostream *` for optional debugging output.